pkg/container/ring_reader.go

```diff
@@ -39,20 +39,29 @@ func NewRingReader(ring *Ring, start uint64) *RingReader {
 }
 
 // Previous reads the event at the current position and decrement the read
-// position by one.
-func (r *RingReader) Previous(ctx context.Context) *v1.Event {
+// position by one. If no event was read, ok is false.
+func (r *RingReader) Previous() (event *v1.Event, ok bool) {
```

Comment:
isn't nil event the same as ok false here? or what's the difference now between nil and false

Reply:
Unfortunately no. The ring may return `nil/true` on `read()` when the buffer is not full and without the `ok` there's no way to tell if one should keep reading or no.

Reply:
I guess what I'm trying to say is we can make this function promise to never return `nil, true` as if everything is fine. Then consumers don't ever have to check for both ok and nil, right?

Reply:
Good point, I'll implement.